crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -976,6 +987,10 @@ unsafe impl<'__w, T: Component> WorldQuery for Ref<'__w, T> {
         world.init_component::<T>()
     }
 
+    fn new_state(components: &Components) -> ComponentId {
+        components.component_id::<T>().unwrap()
```

Comment:
I'd have to unwrap it in the macro instead then. Not sure that's any better.

Reply:
Hmm. Maybe I could use `?` there...

Reply:
Couldn't you just use `?` in the macros?

You'd have to call `.unwrap` or `.expect` somewhere, but doing it in only one place would allow you to give it a better error message. Something like `.expect("This should only return None if the ComponentId has not been initialized. Since this is being called on an initialized QueryState, we know that the components have been initialized.")`

Reply:
completely forgot that you can use `?` on Option for a sec. Definitely feels cleaner now.